apollo-router/src/cache/storage.rs

```diff
@@ -84,25 +88,89 @@ where
 
     pub(crate) async fn get(&self, key: &K) -> Option<V> {
         let mut guard = self.inner.lock().await;
+        let instant_memory = Instant::now();
         match guard.get(key) {
-            Some(v) => Some(v.clone()),
+            Some(v) => {
+                tracing::info!(
+                    monotonic_counter.apollo_router_cache_hit_count = 1u64,
+                    kind = %self.caller,
+                    storage = &tracing::field::display(CacheStorageName::Memory),
+                );
+                let duration = instant_memory.elapsed().as_secs_f64();
+                tracing::info!(
+                    histogram.apollo_router_cache_hit_time = duration,
+                    kind = %self.caller,
+                    storage = &tracing::field::display(CacheStorageName::Memory),
+                );
+                Some(v.clone())
+            }
             #[cfg(feature = "experimental_cache")]
             None => {
+                let instant_redis = Instant::now();
                 if let Some(redis) = self.redis.as_ref() {
                     let inner_key = RedisKey(key.clone());
                     match redis.get::<K, V>(inner_key).await {
                         Some(v) => {
                             guard.put(key.clone(), v.0.clone());
+                            tracing::info!(
+                                monotonic_counter.apollo_router_cache_hit_count = 1u64,
+                                kind = %self.caller,
+                                storage = &tracing::field::display(CacheStorageName::Redis),
+                            );
+                            let duration = instant_redis.elapsed().as_secs_f64();
+                            tracing::info!(
+                                histogram.apollo_router_cache_hit_time = duration,
+                                kind = %self.caller,
+                                storage = &tracing::field::display(CacheStorageName::Redis),
+                            );
                             Some(v.0)
                         }
-                        None => None,
+                        None => {
+                            tracing::info!(
+                                monotonic_counter.apollo_router_cache_miss_count = 1u64,
+                                kind = %self.caller,
+                                storage = &tracing::field::display(CacheStorageName::Redis),
+                            );
+                            let duration = instant_redis.elapsed().as_secs_f64();
+                            tracing::info!(
+                                histogram.apollo_router_cache_miss_time = duration,
+                                kind = %self.caller,
+                                storage = &tracing::field::display(CacheStorageName::Redis),
+                            );
+                            None
+                        }
                     }
                 } else {
+                    tracing::info!(
```

Comment:
if `self.redis` is None, it can be because it was not configured, so we should not generate a cache miss